### PR TITLE
Add tests for recording.api and tag.api

### PR DESCRIPTION
**Pull request by arthurmcgregor**
I've written a simple test for recording.api.test in a similar fashion to user.api.test. 

However, I'm not quite sure what's actually happening or what the point of the test is.... is it actually calling the API or only some mock version? What does the mock do? What sort of errors will this detect?

I stopped when I realised that lines 26 and 45 were going to require me to implement exactly the same procedure as in the actual code, which I thought was to be avoided in tests...

@adbs1 ?

### CODE REVIEW

**Comment by adbs1** [comment]
Here's my thinking:

**The point of the tests:**
- To make us write simple, obvious and testable code.
- Outline the expectations of our code and assert the code meets them.
- Prevent regression by the most efficient means.

**What sort of errors will they detect:**
- They prevent errors getting in at the outset by encouraging the points above.
- Once written they prevent regression and provide confidence that everything works when code is changed, now and every time the tests are run.

Automated e2e or integration tests prevent regression too but in a really inefficient way (extremely slow to run, hard to measure coverage, non-specific so hard to debug, they duplicate testing, require outside services to be running, etc.)

**What does the mock do:**
The fetch mock does nothing other than take a record of what you pass to it and confirm you called it [n] times, your tests assert against that record. Your tests are just mocking the cross-fetch.fetch method ( we don't test the cross-fetch module at all, it has it's own unit tests.) Your method (recordingApi.query) certainly isn't mocked and is called.

With regard to lines 26 and 45, I think they're fine, you're actually testing both the format of the url and that it gets passed to the cross-fetch.fetch method.

Does that help at all?

**Comment by arthurmcgregor** [comment]
OK thanks. Makes more sense. I've written tests for the rest of the recording.api and also the tag.api

File: src/api/Recording.api.test.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  test('query() call with where, limit, offset params', async () =>{
+    let params = {
+      where: testWhere,
+      limit: testLimit,
+      offset: testOffset
+    }
+    let queryString = Object.keys(params).map((key) => {
+      return encodeURIComponent(key) + '=' + encodeURIComponent(params[key]);
+    }).join('&');
+    let expectedUrl = `${Config.api}` + recordingApiUrl + "?" + queryString;
```
Comment (adbs1) [inline src/api/Recording.api.test.js]: `${Config.api}` + recordingApiUrl + "?" + queryString;
Can be simplified to: 
Config.api + recordingApiUrl + "?" + queryString;

File: src/api/Recording.api.test.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      limit: testLimit,
+      offset: testOffset,
+      tagMode: testTagMode,
+      order: testOrder,
+      tags: testTags
+    }
+    let queryString = Object.keys(params).map((key) => {
+      return encodeURIComponent(key) + '=' + encodeURIComponent(params[key]);
+    }).join('&');
+    let expectedUrl = `${Config.api}` + recordingApiUrl + "?" + queryString;
```
Comment (adbs1) [inline src/api/Recording.api.test.js]: As above

File: src/views/RecordingsView.vue [hunk 3]
```diff
@@ -150,19 +156,23 @@ export default {
...
+            return "F/P";
+          } else {
+            return tag.animal;
+          }
+        };
+        let tagString = animal(tags[0]);
+        for (let i = 1; i < tags.length; i ++) {
+          let animalName = animal(tags[i]);
+          tagString = tagString + ', ' + animalName;
         }
```
Comment (adbs1) [inline src/views/RecordingsView.vue]: A leaner option might be(?):

let tagString = '';
tags.forEach(tag => {
  tagString += `, ${animal(tag)}`;
});

**Pull request merged**